src/app/CommandHandler.cpp

```diff
@@ -246,6 +247,20 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
 
     VerifyOrExit(mpCallback->CommandExists(concretePath), err = CHIP_ERROR_INVALID_PROFILE_ID);
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        Access::RequestPath requestPath{ .cluster = concretePath.mClusterId, .endpoint = concretePath.mEndpointId };
+        Access::Privilege requestPrivilege = Access::Privilege::kOperate; // TODO: get actual request privilege
+        err                                = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
+        err                                = CHIP_NO_ERROR; // TODO: remove override
+        if (err != CHIP_NO_ERROR)
+        {
+            ReturnErrorCodeIf(err != CHIP_ERROR_ACCESS_DENIED, err);
```

Comment:
Shouldn't this exit to the error-handling code for this function, instead of returning outright and effectively dropping the error?

Reply:
It returns the error to be handled at a higher level. Any such error (from the access control mechanism that isn't "denied") is effectively a serious error, a malfunctioning of the system. Not sure how we want to handle those, but I can make it respond with status.

Reply:
Done. I still didn't goto the exit label, because it doesn't have good handling of different IM status responses (seems to assume anything is an InvalidCommand) but the effect is the same. We'll want to clean that up later, but as noted, in general a lot of our error handling can be improved.